### PR TITLE
E2E Tests: Add initial set of tests and underlying infra

### PR DESCRIPTION
**Pull request by brbrr**
Adds a dummy test, and code around it. 

### Goals
- Run on master branches
- Run against PRs - should work with multiple PRs CI jobs running at the same time
- Run as part of the multi-stage build process: 1. Run unit tests => create a Jetpack build => Run e2e tests

#### Changes proposed in this Pull Request:

* adds a e2e test suite

#### Testing instructions:

* WIP
* update `WP_BASE_URL` to point to your working Jetpack connected site (together with correct login credentials in `tests/e2e/config/config.js`)
* run `yarn && yarn run test-e2e`
* hope on best 🤞 

#### Proposed changelog entry for your changes:

* no really needed 🤷‍♂

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -163,17 +166,20 @@
...
 		"jest": "24.7.1",
+		"jest-puppeteer": "4.1.1",
 		"lodash": "4.17.11",
 		"mini-css-extract-plugin": "0.5.0",
 		"mocha": "6.1.3",
 		"mockery": "2.1.0",
 		"nock": "10.0.6",
 		"node-wp-i18n": "1.2.3",
 		"prettier": "https://github.com/Automattic/wp-prettier/releases/download/wp-1.17.0/wp-prettier-1.17.0.tgz",
+		"puppeteer": "1.15.0",
```
Comment (simison) [inline package.json]: Noting that this is a bit heavier dependency (discussed p1554884363738500-slack-jetpack-gutenberg).

```bash
$ yarn why puppeteer
...
info Disk size without dependencies: "196.74MB"
info Disk size with unique dependencies: "197.57MB"
info Disk size with transitive dependencies: "199.36MB"
```

I don't think it's soooo heavy that it's worth the hassle to make pulling it optional (https://docs.npmjs.com/files/package.json#optionaldependencies, https://docs.npmjs.com/files/package.json#peerdependencies) but just wanted to give heads up.

File: .travis.yml [hunk 2]
```diff
@@ -47,6 +47,13 @@ matrix:
   - if: branch !~ /(^branch-.*-built)/
     language: node_js
     env: WP_TRAVISCI="yarn build"
+  - name: "E2E tests"
+    php: 7.2
+    env: WP_BRANCH=latest RUN_E2E=1
```
Comment (simison) [inline .travis.yml]: Should you add `WP_MODE=single` for explicit clarity?

Comment (brbrr) [reply .travis.yml]: definitely, but I doubt that this code will be merged to master anyway.  Instead, we would rely on external site configurations such as JN

**Comment by brbrr** [comment]
Closing this in favor of #12403

**Pull request closed without merging**